### DISCUSSION TITLE
Некорректная работа Audiocodes MP-124 (старая ревизия) с MikoPBX

### DISCUSSION THREAD

**Discussion by gurusms** [mikopbx/Core#353] [Q&A]
Есть Audiocodes MP-124(старая ревизия), стоит SIP прошивка  4.60A.016.003 (последняя из возможных).
При работе с MikoPBX 2022.1.99 не воспринимает входящий INVITE, а при исходящем звонке имеем одностороннюю слышимость (RTP поток только в сторону Audiocodes).
MikoPBX и Audiocodes находятся в разных сетях, но маршрутизируемых между собой. Другое оборудование при таком же расположении (в сетях) работает без каких либо проблем.
При использовании оборудования на основе Asterisk 1.6.2.15 при таком же расположении оборудования Audiocodes работает (входящие и исходящие с двусторонним звуковым потоком)
Прилагаю дампы сделанные с  порта Audiocodes. 
other_net_w_miko.pcapng   - дамп с MikoPBX - неудачного входящего звонка и исходящего с односторонней слышимостью 
other_net_w_Aster1.6.pcapng - дамп удачных звонков с оборудованием на Asterisk 1.6.2.15

Так же в дампах можно увидеть собщения Audiocodes на syslog сервер.

В чем может быть проблема при таком поведении Audiocodes с MikoPBX  2022.1.99 ????

[dumps.zip](https://github.com/mikopbx/Core/files/8948074/dumps.zip)

**Comment by boffart** [comment]
На сколько я понял это FXS шлюз. и он регистрируется на MikoPBX на учетных записях sip , которые описываеются в web интерфейсе в разделе "Сотрудники"

Если так, то вероятно проблема в заголовке:
`P-Asserted-Identity: "Lt ······69 ··226 GXP2000" <sip:761118@192.168.142.100> `
PAI красиво работает с аппаратными телефонами Yealink, при переадресациях отображается корректный cid. 
Но вот с устаревшим шлюзом могут быть проблемы, тем более, что в заголовке содержаться какие то не валидные символы, мб кириллица. 

**P-Asserted-Identity** - устанавливается исходя из имени сотрудника. Как вариант попробовать прописать его числовым значением или только латинскими буквами без спец символов. 

На АТС в карточке "**Сотрудника**", в поле "**Дополнительные параметры SIP учетки**" прописать:
```
[endpoint]
send_pai=no
```
это нужно сделать для всех учетных записей, на которых регистрируется шлюз.

**Comment by gurusms** [comment]
При исправлении поля "ФИО сотрудника" для звонящего, это исправило ситуацию со входящими звонками на Audiocodes.
При этом, при входящем на Audiocodes RTP работает в обе стороны и слышимость двусторонняя, т.е. все ОК,
При исходящем с Audiocodes на Мико   RTP работает только в сторону Audiocodes. Слышимость в одну сторону.
Прилагаю дамп таких звонков.

[rtp-bad_to_mikopbx.pcapng.zip](https://github.com/mikopbx/Core/files/8948368/rtp-bad_to_mikopbx.pcapng.zip)

**Reply by boffart** [reply]
dump снят на АТС иди на шлюзе? 
я бы попробовал снять дамп в разных местах:

- на шлюзе
- на АТС
- на маршрутизаторе

это для одного и того же звонка

**Reply by gurusms** [reply]
Все дампы что приложены сняты с зеркалированного порта Audiocodes. Т.е. пакеты непосредственно на Audiocodes.
NAT (трансляция адресов) при маршрутизации не осуществляется.
Немного настораживает ошибка, которую пишет Аудиокодес в syslog после инвайта (с односторонним звуком) 
Syslog message: LOCAL0.WARNING: (lgr_coders_list)(453       ) ?? [WARNING] ACCodersList::IsMatch found match with different intervals g711Alaw64k20 (20 != 150)

Могу сделать дамп sngrep с Мико, если это поможет, но ведь именно в сторону Miko RTP поток и не идет.

**Reply by boffart** [reply]
согласно дампу rtp трафик есть в две стороны. 
если АТС его не получает, то нужно смотреть в какой точке трафик пропадает. вроде в SDP все норм с ip адресами

**Reply by gurusms** [reply]
Все верно. Но в сторону Мико в проблемном звонке уходит всего лишь 2 пакета RTP, далее шлюз шлет вышеуказаную ошибку и RTP поток прекращается. Предполагаю проблема именно с полем Media Attribute (a): maxptime:150 , который присылает Мико в ответе на INVITE:  SIP/SDP	900	Status: 183 Session Progress.
При тех же настройках были проверены кодеки GSM и G.726. На этих кодеках соединение устанавливается двустороннее, и все ОК. Но в моем случае, качество недопустимо низкое.
Проблемы именно с g711Alaw.
Поле  Media Attribute (a): maxptime:150 при соединении с оборудованием на Asterisk 1.6.2.15 отсутствует в ответе сервера.

**Reply by boffart** [reply]
Нашел [обсуждение](https://community.asterisk.org/t/change-maxptime-parameter/70558/3) аналогичное:
https://community.asterisk.org/t/change-maxptime-parameter/70558/3

атрибут "**maxptime**" убрать средствами файлов конфигурации возможности нет. 
нужно править исходники. 

манипулировать можно только ptime атрибутом, и то с ограничениями:
```
[endpoint]
use_ptime=no; опция не понятно на что влияет, но явно связано с ptime
allow = alaw:30; установка ptime для кодека
```


я могу бы попробовать собрать asterisk с подобной правкой
судя по всему изменения касаются только файла `res/res_pjsip_sdp_rtp.c`

сборка кастомного образа MikoPBX займет это 2-3 часа времени, так как задача не популярная, готов предложить только платные работы. к примеру можно добавить опцию disable_max_ptime. 

заявку можете описать на help@miko.ru

**Reply by gurusms** [reply]
Согласно дампу при звонке от Audiocodes rtp поток ограничивается двумя пакетами. При этом настораживает ошибка "Match found match with different intervals g711Alaw64k20 (20 != 150)" которая возникает исключительно при односторонней связи при исходящем звонке. При этом настораживает реакция шлюза на Media Attribute (a): maxptime:150 с этой ошибки и в RTSP пакете [Stream setup by SDP (frame 15)). При нормальной связи обычно frame 410.
Возможно все это не имеет существенного значения.
И все же при той же конфигурации оборудования, в сети были попробованы кодек GSM и G726 и на этих кодеках проблем таких с односторонней связью не было. Т.е. связь была в обе стороны. Но к сожалению качество звука недопустимое в моем случае.
Так же был проверен вариант на FreePBX 5 на Asterisk (Ver. 1.8.32.1). И так же по звуку не было никаких проблем, при звонках в обе стороны мы получали двустороннюю связь.
К сожалению разобраться с этой непростой задачей очень сложно. И приходится констатировать несовместимость оборудования Audiocodes и последней версии MikoPBX. Впрочем на версии MikoPBX 2019.4.53 эта ситуация с односторонним вызовом - так же проявлялась с односторонней слышимостью.

**Reply by gurusms** [reply]
Спасибо за внимание. Вопрос в целом прояснен о несовместимости этого старого оборудования. Для этой железки мы решим вопрос обходными путями, без использования MikoPBX, к сожалению.
